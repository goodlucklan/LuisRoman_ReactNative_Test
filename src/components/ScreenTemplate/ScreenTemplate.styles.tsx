import { StyleSheet } from "react-native";
import { $palette } from "../../utils/variables";

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: $palette.white,
  },
  scrollView: {
    flexGrow: 1,
    backgroundColor: $palette.white,
  },
});
