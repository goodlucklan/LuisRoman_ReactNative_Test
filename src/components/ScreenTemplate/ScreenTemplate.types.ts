// Interfaces and types from component ScreenTemplate

// Component Props
export interface ScreenTemplateProps {
  scrollable?: boolean;
  children?: any;
}
