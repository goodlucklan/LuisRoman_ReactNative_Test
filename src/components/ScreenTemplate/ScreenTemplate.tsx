import React from "react";
import { View, ScrollView } from "react-native";

import { styles } from "./ScreenTemplate.styles";
import { ScreenTemplateProps as Props } from "./ScreenTemplate.types";

const ScreenTemplate: React.FC<Props> = (props) => {
  const { scrollable = true, children } = props;
  if (!scrollable) {
    return <View style={[styles.container]}>{children}</View>;
  } else {
    <View style={[styles.container]}>
      <ScrollView
        contentContainerStyle={styles.scrollView}
        showsVerticalScrollIndicator={false}
      >
        {children}
      </ScrollView>
    </View>;
  }
};

ScreenTemplate.defaultProps = {};

export default ScreenTemplate;
