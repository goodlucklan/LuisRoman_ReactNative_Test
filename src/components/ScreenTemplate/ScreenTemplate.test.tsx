import React from "react";
import ScreenTemplate from "./ScreenTemplate";

describe("ScreenTemplate Component", () => {
  test("Render a global Component", () => {
    expect(<ScreenTemplate />).toMatchSnapshot();
  });
});
