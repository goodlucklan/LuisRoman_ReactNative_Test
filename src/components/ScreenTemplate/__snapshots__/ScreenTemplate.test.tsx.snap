// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenTemplate Component Render a global Component 1`] = `<ScreenTemplate />`;
